hw/bsp/stm32h7rs/linker/stm32h7s3xx_flash.ld

```diff
@@ -196,6 +197,11 @@ SECTIONS
     . = ALIGN(8);
   } >DTCM
 
+  .dtcm_data :
```

Comment:
can we remove this, and use the `noncacheable_buffer` instead, I like to keep thing closed to stock stm32 linker as possible.

Reply:
wait, does the current code configure this section as non-cacheable, while noncacheable_buffer section need extra works, or both of them need to be configured the same way ?

Reply:
`dtcm_data` is only used by RTT to have a non-cached buffer, DWC2 doesn't relied on this. 

I've added `SCB_EnableDCache` for benchmark only, to keep it simple we can also keep the DCache disabled.

Reply:
so it is possible to use the default noncacheable_buffer section instead of adding new one. That would allow us to use the stock IAR linker as well ? You may want to pull first, since, I push the fix for the linker issue with clang (need to have memory value before provide, and drop the READONLY keywords)